javaparser-core/src/main/java/com/github/javaparser/ast/type/TypeParameter.java

```diff
@@ -72,8 +73,13 @@ public TypeParameter(final String name, final NodeList<ClassOrInterfaceType> typ
         this(null, new SimpleName(name), typeBound, new NodeList<>());
     }
 
+    /**
+     * @deprecated range shouldn't be in utility constructors.
```

Comment:
This is a utility constructor for the grammar, I made it pick the usual longer constructor instead.

Why no range in the normal constructors? Because it seems to be a marginal use case to me, shuffling nodes around. Am I wrong?